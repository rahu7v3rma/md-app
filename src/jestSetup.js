import mockRNCameraRoll from '@react-native-camera-roll/camera-roll/src/__mocks__/nativeInterface';
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock';
import mockRNDeviceInfo from 'react-native-device-info/jest/react-native-device-info-mock';

import { MockData } from './utils/mockData';

// setImmediate and clearImmediate aren't mocked by useFakeTimers if they are
// not set to a function (https://github.com/jestjs/jest/pull/11599)
global.setImmediate = jest.fn();
global.clearImmediate = jest.fn();
jest.useRealTimers();

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

jest.mock('react-native-keychain', () => {
    return {
        SECURITY_LEVEL_ANY: 'MOCK_SECURITY_LEVEL_ANY',
        SECURITY_LEVEL_SECURE_SOFTWARE: 'MOCK_SECURITY_LEVEL_SECURE_SOFTWARE',
        SECURITY_LEVEL_SECURE_HARDWARE: 'MOCK_SECURITY_LEVEL_SECURE_HARDWARE',
        setGenericPassword: jest.fn().mockResolvedValue(),
        getGenericPassword: jest.fn().mockResolvedValue(false),
        resetGenericPassword: jest.fn().mockResolvedValue()
    };
});

jest.mock('react-native-splash-screen', () => {
    return {
        hide: jest.fn()
    };
});

export let mockUnwrap = jest.fn().mockImplementation(() => {
    return Promise.resolve({});
});
export const mockDispatch = jest.fn().mockImplementation(() => {
    return {
        unwrap: mockUnwrap
    };
});
jest.mock('@/hooks', () => ({
    useAppDispatch: () => mockDispatch,
    useChannelPreviewInfo: () => mockDispatch
}));
export const mockChannelId = 'mockChannelId';
export const mockChatClient = jest.fn().mockImplementation(() => ({
    userID: 1,
    chat_id: 1,
    queryChannels: jest.fn().mockImplementation(() => {
        return Promise.resolve([
            {
                cid: 99
            }
        ]);
    }),
    activeChannels: {
        mockChannelId
    }
}));
jest.mock('@/contexts/appChat', () => ({
    useAppChat: () => {
        return {
            chatClient: mockChatClient(),
            setActiveChatChannel: jest.fn()
        };
    }
}));
export const mockRNSha256 = jest
    .fn()
    .mockImplementation(() => Promise.resolve('123'));
jest.mock('react-native-sha256', () => {
    return {
        sha256: mockRNSha256
    };
});

jest.mock('react-native-share', () => ({
    default: jest.fn()
}));

jest.mock('@/services/api', () => ({
    getImageUploadCredentials: jest.fn(() => ({
        keyPrefix: 'keyPrefix',
        fileName: 'fileName',
        bucketName: 'bucketName'
    }))
}));

jest.mock('@/reducers/content', () => ({
    ContentSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        journey: MockData.journey,
        navigationKey: 'Main.Home',
        activeBlock: MockData.activeBlock,
        activeLesson: {
            lesson: {}
        },
        journeyBlocks: MockData.journeyBlocks
    })),
    loadJourney: jest.fn(),
    getActiveLesson: jest.fn()
}));

export const mockChatProfile = jest.fn(() => ({
    apiKey: 'apiKey',
    token: 'token',
    userId: 'userId',
    firebasePushProviderName: 'firebasePushProviderName'
}));
export const mockDeleteAccount = jest.fn();
jest.mock('@/reducers/user', () => ({
    UserSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        userInfo: {
            username: 'user1',
            first_name: 'Some',
            last_name: 'Name',
            email: 'user1@example.example'
        },
        userProfile: {
            diabetes_type: 'Prediabetes',
            image: null,
            onboarding_form_url: null
        },
        coach: {
            first_name: 'name',
            profile_image: 'profile_image'
        },
        chatProfile: mockChatProfile(),
        hasCoachChat: false
    })),
    userLogin: jest.fn(),
    changePassword: jest.fn(),
    resetPassword: jest.fn(),
    resetPasswordConfirm: jest.fn(),
    resetPasswordVerify: jest.fn(),
    getCoach: jest.fn(),
    refreshProfileSession: jest.fn(),
    getProfile: jest.fn(),
    searchFilterAction: jest.fn(),
    clearSearchText: jest.fn(),
    deleteAccount: mockDeleteAccount
}));
export const mockRNFSExists = jest
    .fn()
    .mockImplementation(() => Promise.resolve(false));
export const mockRNFSCachesDirectoryPath = 'mockCachesDirectoryPath';
export const mockRNFSDownloadFile = jest.fn().mockImplementation(() => ({
    promise: Promise.resolve({ statusCode: 200 })
}));
jest.mock('react-native-fs', () => {
    return {
        readFile: jest.fn(),
        exists: mockRNFSExists,
        CachesDirectoryPath: mockRNFSCachesDirectoryPath,
        downloadFile: mockRNFSDownloadFile,
        unlink: jest.fn().mockImplementation(() => Promise.resolve())
    };
});
jest.mock('react-native-image-crop-picker', () => {
    return {
        openPicker: jest.fn(),
        launchImageLibrary: jest.fn(),
        openCamera: jest.fn()
    };
});

jest.mock('@react-native-camera-roll/camera-roll', () => mockRNCameraRoll);
jest.mock('react-native-reanimated', () => ({
    ...require('react-native-reanimated/mock')
}));
jest.mock('react-native-device-info', () => mockRNDeviceInfo);
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');

jest.mock('@/reducers/track', () => ({
    TrackSelectors: jest.fn().mockImplementation(() => ({
        trackFastState: '',
        trackStartedAt: '',
        trackTimeLimitInSeconds: 57600,
        trackedTimeInSeconds: 0
    })),
    trackStart: jest.fn(),
    trackUpdate: jest.fn()
}));

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo);

jest.mock('@notifee/react-native', () =>
    require('@notifee/react-native/jest-mock')
);

export const mockOnTokenRefresh = jest.fn();
jest.mock('@react-native-firebase/messaging', () => ({
    __esModule: true,
    default: () => ({ onTokenRefresh: mockOnTokenRefresh }),
    FirebaseMessagingTypes: jest.fn()
}));

export const mockNavigate = jest.fn();
export const mockPop = jest.fn();
export const mockGoBack = jest.fn();
export const mockReplace = jest.fn();
export const mockNavigationDispatch = jest.fn();
export const mockUseRoute = jest.fn().mockImplementation(() => ({
    params: {
        id: 1,
        unit: 'kg',
        code: '11111',
        url: 'https://www.google.com/'
    }
}));

jest.mock('@/services/notification', () => ({
    resetNotifications: jest.fn(),
    registerBackgroundNotificationsListener: jest.fn(),
    getFCMToken: jest.fn().mockImplementation(() => {
        return Promise.resolve({});
    }),
    triggerFirebaseForegroundNotification: jest.fn(),
    registerFirebaseMessageListener: jest.fn(),
    requestNotificationPermissions: jest.fn()
}));

jest.mock('@react-navigation/native', () => {
    const real = jest.requireActual('@react-navigation/native');
    return {
        ...real,
        useNavigation: jest.fn().mockImplementation(() => {
            return {
                canGoBack: jest.fn().mockImplementation(() => {
                    return true;
                }),
                goBack: mockGoBack,
                navigate: mockNavigate,
                pop: mockPop,
                replace: mockReplace,
                dispatch: mockNavigationDispatch
            };
        }),
        useRoute: mockUseRoute,
        useIsFocused: jest.fn().mockImplementation(() => ({
            isFocused: true
        })),
        useFocusEffect: jest.fn()
    };
});

export const mockToast = jest.fn();

jest.mock('react-native-toast-message', () => ({
    show: mockToast
}));

export const mockGetChannelDisplayImage = jest.fn();
jest.mock('@/services/chat', () => ({
    getChannelDisplayImage: mockGetChannelDisplayImage
}));

export const mockConnectUser = jest.fn(() => Promise.resolve());
export const mockDisconnectUser = jest.fn();
export const mockGetInstance = jest.fn(() => ({
    connectUser: mockConnectUser,
    disconnectUser: mockDisconnectUser
}));
jest.mock('stream-chat', () => ({
    StreamChat: {
        getInstance: mockGetInstance
    }
}));

jest.mock('@aws-sdk/client-s3', () => ({
    S3Client: jest.fn(() => ({
        send: jest.fn(() => Promise.resolve())
    })),
    PutObjectCommand: jest.fn().mockImplementation(() => {
        return {};
    })
}));

jest.mock('react-native-pdf', () => null);
