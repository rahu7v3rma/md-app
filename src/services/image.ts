import { Buffer } from 'buffer';

import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
import { fromWebToken } from '@aws-sdk/credential-providers';
import ImagePicker from 'react-native-image-crop-picker';

// these two imports are needed for aws-sdk to work properly
import 'react-native-get-random-values';
import 'react-native-url-polyfill/auto';

import { getImageUploadCredentials } from '@/services/api';

const pickImage = async (props: {
    type: 'camera' | 'gallery';
    height?: number;
    width?: number;
    cropping?: boolean;
}) => {
    const { type, height, width, cropping } = props;
    const imagePickerMethod =
        type === 'camera' ? ImagePicker.openCamera : ImagePicker.openPicker;
    return await imagePickerMethod({
        height: height || 200,
        width: width || 200,
        cropping,
        cropperCircleOverlay: height ? false : true,
        includeBase64: true
    });
};

const encodeImage = (
    base64Content: string,
    key: string
): Promise<Uint8Array> => {
    return new Promise((resolve) => {
        let finalKey = key;

        // while the key length is under 64 characters append the original key
        // over and over
        while (finalKey.length < 64) {
            finalKey += key;
        }

        // shorten key to be exactly 64 characters long
        if (finalKey.length > 64) {
            finalKey = finalKey.substring(0, 64);
        }

        const content = Buffer.from(base64Content, 'base64');

        /* eslint-disable no-bitwise */
        resolve(
            content.map(
                (uint, i) => uint ^ finalKey.charCodeAt(i % finalKey.length)
            )
        );
    });
};

const uploadImage = (
    base64Content: string,
    contentType: string,
    username: string,
    publicImage: boolean
): Promise<string> => {
    return Promise.all([
        getImageUploadCredentials(publicImage),
        publicImage
            ? Promise.resolve(Buffer.from(base64Content, 'base64'))
            : encodeImage(base64Content, username)
    ]).then(([uploadCredentialsResponse, encodedImage]) => {
        const imagePath =
            uploadCredentialsResponse.keyPrefix +
            uploadCredentialsResponse.fileName;

        // `mock` will only be returned when working with a mocked service.
        // the real service will never return this field
        if (uploadCredentialsResponse.mock) {
            return imagePath;
        }

        const client = new S3Client({
            region: uploadCredentialsResponse.regionName,
            credentials: fromWebToken({
                webIdentityToken: uploadCredentialsResponse.token,
                roleArn: uploadCredentialsResponse.roleArn
            })
        });

        return client
            .send(
                new PutObjectCommand({
                    Bucket: uploadCredentialsResponse.bucketName,
                    Key: imagePath,
                    Body: encodedImage,
                    ContentType: contentType
                })
            )
            .then((_putOutput) => imagePath);
    });
};

export { pickImage, uploadImage };
