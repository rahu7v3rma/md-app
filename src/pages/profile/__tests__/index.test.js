import React from 'react';
import 'react-native';
import { resetGenericPassword } from 'react-native-keychain';
import { Provider } from 'react-redux';
import renderer, { act } from 'react-test-renderer';
import { PersistGate } from 'redux-persist/integration/react';

import { navigationRef } from '@/navigation';
import store, { persistor } from '@/store';

import { mockNavigate, mockToast } from '../../../jestSetup';
import imageMock from '../../../pages/logMeal/__mocks__/image.json';
import Profile from '../../../pages/profile';
import { userLogin } from '../../../reducers/user';
import * as imageServices from '../../../services/image';

jest.unmock('@/reducers/user');

jest.unmock('@/services/api');
jest.mock('@/services/api', () => ({
    login: jest.fn().mockReturnValue({
        email: 'user1@example.example',
        first_name: 'Some',
        last_name: 'Name',
        username: 'user1'
    })
}));

jest.mock('@/navigation', () => ({
    navigationRef: {
        dispatch: jest.fn()
    }
}));

jest.mock('react-native-keychain', () => {
    return {
        setGenericPassword: jest.fn(() => {}),
        getGenericPassword: jest.fn(() => {}),
        resetGenericPassword: jest.fn(() => {})
    };
});

jest.spyOn(imageServices, 'pickImage').mockResolvedValue(imageMock);
jest.spyOn(imageServices, 'uploadImage').mockResolvedValue(
    'validUploadedImagePath'
);

it('logout a user', async () => {
    expect(store.getState().user.userInfo).toBeNull();
    await store.dispatch(
        userLogin({ email: 'test@test.com', password: 'test_pass' })
    );
    expect(store.getState().user.userInfo).not.toBeNull();
    const tree = renderer.create(
        <PersistGate loading={null} persistor={persistor}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </PersistGate>
    );
    let element = tree.root.findByProps({ testID: 'logout' }).props;
    expect(resetGenericPassword).not.toBeCalled();
    expect(navigationRef.dispatch).not.toBeCalled();
    await act(() => element.onPress());
    expect(resetGenericPassword).toBeCalled();
    expect(navigationRef.dispatch).toBeCalledWith({
        type: 'RESET',
        payload: { index: 0, routes: [{ name: 'SignIn' }] }
    });
    expect(store.getState().user.userInfo).toBeNull();
});

it('profile snapshot', () => {
    const tree = renderer.create(
        <PersistGate loading={null} persistor={persistor}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </PersistGate>
    );
    expect(tree).toMatchSnapshot();
});

it('user can change password', async () => {
    const tree = renderer.create(
        <PersistGate loading={null} persistor={persistor}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </PersistGate>
    );
    let element = tree.root.findByProps({ testID: 'changepassword' }).props;
    await act(() => element.onPress());
    expect(mockNavigate).toBeCalledWith('ChangePassword', {});
});

it('user can delete account', async () => {
    const tree = renderer.create(
        <PersistGate loading={null} persistor={persistor}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </PersistGate>
    );
    let element = tree.root.findByProps({ testID: 'deleteAccount' }).props;
    await act(() => element.onPress());
    expect(mockNavigate).toBeCalledWith('DeleteAccount');
});

it('user can change profile picture by choose image from storage', async () => {
    await act(async () => {
        expect(store.getState().user.userInfo).toBeNull();
        await store.dispatch(
            userLogin({ email: 'test@test.com', password: 'test_pass' })
        );
        expect(store.getState().user.userInfo).not.toBeNull();
    });
    const tree = renderer.create(
        <PersistGate loading={null} persistor={persistor}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </PersistGate>
    );

    const editProfilePic = tree.root.findByProps({
        testID: 'editProfilePic'
    });
    await act(() => editProfilePic.props.onPress());

    const choosePhotoButton = tree.root.findByProps({
        testID: 'choosePhoto'
    });
    await act(() => choosePhotoButton.props.onPress());
    expect(imageServices.uploadImage).toBeCalledWith(
        imageMock.data,
        imageMock.mime,
        'user1',
        true
    );
});

it('user can change profile picture by take image from camera', async () => {
    const tree = renderer.create(
        <PersistGate loading={null} persistor={persistor}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </PersistGate>
    );

    const editProfilePic = tree.root.findByProps({
        testID: 'editProfilePic'
    });
    await act(() => editProfilePic.props.onPress());

    const takePhotoButton = tree.root.findByProps({
        testID: 'takePhoto'
    });
    await act(() => takePhotoButton.props.onPress());
    expect(imageServices.uploadImage).toBeCalledWith(
        imageMock.data,
        imageMock.mime,
        'user1',
        true
    );
});

describe('test photos permission', () => {
    it('when there is no photos permission then display toast', async () => {
        jest.spyOn(imageServices, 'pickImage').mockRejectedValue({
            message: '',
            code: 'E_NO_LIBRARY_PERMISSION',
            name: 'Error'
        });
        await act(async () => {
            await store.dispatch(
                userLogin({ email: 'test@test.com', password: 'test_pass' })
            );
            expect(store.getState().user.userInfo).not.toBeNull();
        });
        const tree = renderer.create(
            <PersistGate loading={null} persistor={persistor}>
                <Provider store={store}>
                    <Profile />
                </Provider>
            </PersistGate>
        );

        const editProfilePic = tree.root.findByProps({
            testID: 'editProfilePic'
        });
        await act(() => editProfilePic.props.onPress());

        const choosePhotoButton = tree.root.findByProps({
            testID: 'choosePhoto'
        });
        await act(() => choosePhotoButton.props.onPress());
        expect(mockToast).toBeCalledWith({
            type: 'info',
            text1: "To use a photo, go to your phone's settings and allow library access",
            position: 'bottom'
        });
    });
});

describe('test camera permission', () => {
    it('when there is no camera permission then display toast', async () => {
        jest.spyOn(imageServices, 'pickImage').mockRejectedValue({
            message: '',
            code: 'E_NO_CAMERA_PERMISSION',
            name: 'Error'
        });
        await act(async () => {
            await store.dispatch(
                userLogin({ email: 'test@test.com', password: 'test_pass' })
            );
            expect(store.getState().user.userInfo).not.toBeNull();
        });
        const tree = renderer.create(
            <PersistGate loading={null} persistor={persistor}>
                <Provider store={store}>
                    <Profile />
                </Provider>
            </PersistGate>
        );

        const editProfilePic = tree.root.findByProps({
            testID: 'editProfilePic'
        });
        await act(() => editProfilePic.props.onPress());

        const takePhotoButton = tree.root.findByProps({
            testID: 'takePhoto'
        });

        await act(() => takePhotoButton.props.onPress());

        expect(mockToast).toBeCalledWith({
            type: 'info',
            text1: "To take a photo, go to your phone's settings and allow camera access",
            position: 'bottom'
        });
    });
});
