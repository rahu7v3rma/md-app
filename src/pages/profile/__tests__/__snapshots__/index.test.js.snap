// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FAFAFF",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 22,
          "paddingHorizontal": 19,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "width": "20%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#D3E29F40",
              "borderRadius": 100,
              "height": 48,
              "justifyContent": "center",
              "margin": 0,
              "opacity": 1,
              "paddingHorizontal": 15.5,
              "paddingVertical": 24,
              "shadowColor": "",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "width": 48,
            }
          }
          testID="leftButton"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={14}
            bbWidth={8}
            fill="none"
            focusable={false}
            height={14}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "height": "auto",
                  "width": "auto",
                },
                {
                  "flex": 0,
                  "height": 14,
                  "width": 8,
                },
              ]
            }
            vbHeight={14}
            vbWidth={8}
            width={8}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M5.97999 0.319993L2.76999 3.52999L0.79999 5.48999C-0.0300097 6.31999 -0.0300097 7.66999 0.79999 8.49999L5.97999 13.68C6.65999 14.36 7.81999 13.87 7.81999 12.92V7.30999V1.07999C7.81999 0.119992 6.65999 -0.360007 5.97999 0.319993Z"
                fill={4280752721}
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "width": "60%",
          }
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#72996B",
                "fontSize": 16,
                "fontWeight": "600",
                "lineHeight": 24,
              },
            ]
          }
          testID="title"
        >
          Settings
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "flex-end",
            "width": "20%",
          }
        }
      />
    </View>
    <View
      style={false}
    >
      <View
        style={false}
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      style={
        {
          "marginTop": 45,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/images/default_avatar.jpg",
            }
          }
          style={
            {
              "borderRadius": 50,
              "height": 110,
              "width": 110,
            }
          }
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#F0E957",
            "borderColor": "white",
            "borderRadius": 18,
            "borderWidth": 1,
            "bottom": -2,
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "right": -2,
            "width": 36,
          }
        }
        testID="editProfilePic"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={18}
          bbWidth={18}
          fill="none"
          focusable={false}
          height={18}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 18,
                "width": 18,
              },
            ]
          }
          vbHeight={20}
          vbWidth={20}
          width={18}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M7.93 8.806l3.516-3.515a2.307 2.307 0 113.263 3.263l-3.515 3.516a11.925 11.925 0 01-5.54 3.136l-.68.17a.289.289 0 01-.35-.35l.17-.68a11.924 11.924 0 013.137-5.54z"
              fill={4294967295}
              fillOpacity={0.24}
              propList={
                [
                  "fill",
                  "fillOpacity",
                ]
              }
            />
            <RNSVGPath
              d="M11.446 5.291l.53.53-.53-.53zM7.93 8.806l.53.53-.53-.53zm6.778-.252l-.53-.53.53.53zm-3.515 3.516l-.53-.53.53.53zm-5.54 3.136l-.182-.728.182.728zm-.68.17l.183.727-.182-.727zm-.35-.35l.728.181-.728-.182zm.17-.68l.728.182-.728-.182zm6.121-9.585L7.4 8.276l1.06 1.06 3.516-3.514-1.06-1.061zm3.263 3.263l-3.515 3.515 1.061 1.06 3.515-3.514-1.06-1.06zm-8.706 6.454l-.68.17.365 1.455.68-.17-.365-1.455zm-.12.73l.17-.68-1.456-.364-.17.68 1.456.363zm-.56-.56a.461.461 0 01.56.56l-1.455-.365c-.19.761.499 1.45 1.26 1.26l-.364-1.455zm5.871-3.109a11.173 11.173 0 01-5.19 2.94l.363 1.455a12.674 12.674 0 005.888-3.334l-1.06-1.06zM7.4 8.276a12.674 12.674 0 00-3.334 5.888l1.456.364A11.174 11.174 0 018.46 9.337L7.4 8.277zm5.677-4.41c-.81 0-1.588.322-2.162.895l1.06 1.06a1.557 1.557 0 011.102-.456v-1.5zm1.558 3.057c0 .413-.165.809-.457 1.101l1.061 1.06a3.057 3.057 0 00.896-2.161h-1.5zm1.5 0a3.057 3.057 0 00-3.058-3.058v1.5c.86 0 1.558.698 1.558 1.558h1.5zM13.8 8.513c-.431.144-1.098-.052-1.68-.634-.582-.582-.778-1.249-.634-1.68l-1.423-.475c-.4 1.2.22 2.438.997 3.215.777.778 2.014 1.397 3.215.997L13.8 8.513z"
              fill={4294967295}
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "marginTop": 32,
            "textAlign": "center",
          },
          {
            "color": "#4D4F4D",
            "fontSize": 28,
            "fontWeight": "700",
            "lineHeight": 36,
          },
        ]
      }
    >
      undefined undefined
    </Text>
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "textAlign": "center",
          },
          {
            "color": "#72996B",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
        ]
      }
    />
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "flex-end",
          "marginBottom": 70,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#72996B",
            "borderBottomWidth": 0,
            "borderColor": "#72996B",
            "borderRadius": 20,
            "borderWidth": 0,
            "justifyContent": "center",
            "margin": 0,
            "opacity": 1,
            "paddingHorizontal": 15.5,
            "paddingVertical": 24,
            "width": "100%",
          }
        }
        testID="changepassword"
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 22,
              },
            ]
          }
        >
          Change password
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderBottomWidth": 0,
            "borderColor": "#72996B",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0.1,
            "justifyContent": "center",
            "margin": 0,
            "marginTop": 16,
            "opacity": 1,
            "paddingHorizontal": 15.5,
            "paddingVertical": 24,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.05,
            "shadowRadius": 17,
            "width": "100%",
          }
        }
        testID="logout"
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#72996B",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 22,
              },
            ]
          }
        >
          Logout
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#B16663",
            "borderBottomWidth": 0,
            "borderColor": "#72996B",
            "borderRadius": 20,
            "borderWidth": 0,
            "justifyContent": "center",
            "margin": 0,
            "marginTop": 16,
            "opacity": 1,
            "paddingHorizontal": 15.5,
            "paddingVertical": 24,
            "width": "100%",
          }
        }
        testID="deleteAccount"
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              undefined,
              {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 22,
              },
            ]
          }
        >
          Delete account
        </Text>
      </View>
    </View>
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "bottom": 20,
            "position": "absolute",
          },
          {
            "color": "#A4AAAF",
            "fontSize": 12,
            "fontWeight": "500",
            "lineHeight": 20,
          },
        ]
      }
    >
      Version app 
      unknown
    </Text>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
